saul-examples/src/main/scala/edu/illinois/cs/cogcomp/saulexamples/nlp/EntityRelation/EntityRelationConstrainedClassifiers.scala

```diff
@@ -6,43 +6,44 @@
   */
 package edu.illinois.cs.cogcomp.saulexamples.nlp.EntityRelation
 
-import edu.illinois.cs.cogcomp.infer.ilp.OJalgoHook
-import edu.illinois.cs.cogcomp.saul.classifier.ConstrainedClassifier
+import edu.illinois.cs.cogcomp.saul.classifier.infer.{ ConstrainedClassifier, OJAlgo }
 import edu.illinois.cs.cogcomp.saulexamples.EntityMentionRelation.datastruct.{ ConllRawToken, ConllRelation }
-import edu.illinois.cs.cogcomp.saulexamples.nlp.EntityRelation.EntityRelationClassifiers._
-import edu.illinois.cs.cogcomp.saulexamples.nlp.EntityRelation.EntityRelationConstraints._
 
 object EntityRelationConstrainedClassifiers {
-  val erSolver = new OJalgoHook
-
-  object OrgConstrainedClassifier extends ConstrainedClassifier[ConllRawToken, ConllRelation](OrganizationClassifier) {
-    def subjectTo = relationArgumentConstraints
-    override val pathToHead = Some(-EntityRelationDataModel.pairTo2ndArg)
+  object OrgConstrainedClassifier extends ConstrainedClassifier[ConllRawToken, ConllRelation] {
+    override lazy val onClassifier = EntityRelationClassifiers.OrganizationClassifier
+    override def pathToHead = Some(-EntityRelationDataModel.pairTo2ndArg)
+    override def subjectTo = Some(EntityRelationConstraints.relationArgumentConstraints)
```

Comment:
> What if I want to say subjectTo = Some(worksForConstraint)? what will be the syntax? workForContatint needs an input parameter. 

Regarding the first questions ([as also mentioned in the documentation](https://github.com/danyaljj/saul-1/blob/4bce6b79093b563e5a5df52a840d4af17c9e5b64/saul-core/doc/SAULLANGUAGE.md#collection-operators)), the definition of the constraints starts with `Node` and use `ForEach` operator. If you want to define "worksForConstraints", instead of having it as a function: 
```scala 
  // if x is works-for relation, then its first argument should be person, and second argument should be organization.
  def worksForConstraint(x: ConllRelation) = {
    (WorksForClassifier on x isTrue) ==> ((PersonClassifier on x.e1 isTrue) and (OrganizationClassifier on x.e2 isTrue))
  }
``` 
it should be written as 

```scala 
  // if x is works-for relation, then its first argument should be person, and second argument should be organization.
  def worksForConstraint = EntityRelationDataModel.pairs.ForEach { x: ConllRelation => 
    (WorksForClassifier on x isTrue) ==> ((PersonClassifier on x.e1 isTrue) and (OrganizationClassifier on x.e2 isTrue))
  }
```


> Another relevant comment, how can I just express true or false i.e. constant expressions as constraints. Please add these to the documentation also. 

You can't (and you shouldn't) define constants.